src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2660,7 +2662,19 @@ MultiRouterPlannableQuery(Query *query, RelationRestrictionContext *restrictionC
 			char partitionMethod = PartitionMethod(distributedTableId);
 
 			if (!(partitionMethod == DISTRIBUTE_BY_HASH || partitionMethod ==
-				  DISTRIBUTE_BY_NONE))
+				  DISTRIBUTE_BY_NONE || partitionMethod == DISTRIBUTE_BY_RANGE))
+			{
+				return false;
+			}
+
+			hasHashDistributedTable |= partitionMethod == DISTRIBUTE_BY_HASH;
```

Comment:
We don't allow joins between different distribution types in our real-time planner/executor. This kind of joins are handled by repartitioning. 

I do not see a value allowing some of them in router execution. The same query could become magically supported and not supported between 2 setups (test and prod).